plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -733,6 +733,10 @@ public function set_minimum_amount( $amount ) {
 	 * @param float $amount Maximum amount.
 	 */
 	public function set_maximum_amount( $amount ) {
+		if ( $amount > 0 && $this->get_minimum_amount() > $amount ) {
+			$this->error( 'coupon_invalid_maximum_amount', __( 'Invalid Maximum spend value.', 'woocommerce' ) );
```

Comment:
The message should be more specific to the check happening above.

`Maximum spend value cannot be greater than the minimum value` or something similar

Reply:
Let's consider this as part of a review of all the validation errors in that page.